src/Hamiltonians/TwoBodyCorrelation.jl

```diff
@@ -0,0 +1,70 @@
+# Methods that need to be implemented:
+#   •  starting_address(::AbstractHamiltonian) - not needed
+"""
+    TwoBodyCorrelation(d::Int)
+
+Implements the two-body correlation operator for `BoseFS2C` for computing the correlation function
+between the impurity and a boson from the Bose gas as a function of the distance `d`, which can take
+`Integer` values from `0` to `M` (the number of modes).
+It works on Fock states in momentum space and could return a `Complex` value.
+
+```math
+\\hat{G}_2^{(d)} = \\frac{1}{M}\\sum_{spqr}^M e^{-id(p-q)2π/M} a^†_{s} b^†_{p}  b_q a_r δ_{s+p,q+r}
+```
+
+# Arguments
+
+* `d`: the distance between the impurity and a boson from the Bose gas.
+
+# See also
+
+* [`BoseHubbardMom1D2C`](@ref)
+* [`BoseFS2C`](@ref)
+
+"""
+struct TwoBodyCorrelation <: AbstractHamiltonian{ComplexF64}
+    d::Int
+end
+
+function num_offdiagonals(g::TwoBodyCorrelation, add::BoseFS2C)
+    M = num_modes(add)
+    sa = numberoccupiedsites(add.bsa)
+    sb = numberoccupiedsites(add.bsb)
+    return sa*(M-1)*sb
+    # number of excitations that can be made
+end
+
+"""
+    diagonal_element(g::TwoBodyCorrelation, add::BoseFS2C{NA,NB,M,AA,AB})
+
+The diagonal element in [`TwoBodyCorrelation`](@ref), where `(p-q)=0`, hence
+it becomes
+```math
+\\frac{1}{M}\\sum_{spqr}^M a^†_{k} b^†_{p}  b_p a_k .
```

Comment:
This formula does not look right. Should `d` appear in there?  Other indices should be summed over.

Reply:
Fixed now.